src/main.rs

```diff
@@ -51,16 +51,19 @@ fn main() ->  Result<(), Err()> {
 
     // Send the email
     let response = Client::new()
-        .default_headers(headers)
-        .json(&body)
         .post("https://api.sendgrid.com/v3/mail/send")
-        .send().await?;
+        .json(&body)
+        .header(header::AUTHORIZATION, format!("Bearer {}", api_key))
+        .header(header::CONTENT_TYPE, header::HeaderValue::from_static("application/json"))
+        .send()
+        .await
+        .unwrap();
 
     // Handle/Check the response
-    match response.status() {
+    match response.status().as_u16() {
```

Comment:
.status() returns a StatucCode, not an "integer", using "as_u116()" converts StatusCode into an integer ( u16 )